From e167cd098156bca099d5b695a67f369942c2ccda Mon Sep 17 00:00:00 2001
From: ywan171 <yi.a.wang@intel.com>
Date: Sat, 28 Jul 2012 00:30:41 +0800
Subject: [PORT FROM MAIN]NuPlayerRender: Fix uint32_t numFramesPendingPlayout overrun issue

BZ: 48710

Add protection to avoid uint32_t numFramesPendingPlayout overrun.
If overrun happens, the video will be post a large delay time casuing video frozen.

orignal patch
43855: orig-bz(BZ: 27202)

Change-Id: I9e9bb8c90c4c12f35f49857b69c70c7078a01b05
Orig-Change-Id: I0fc28327de44837845e0052a3269760ba0d05f1c
Signed-off-by: ywan171 <yi.a.wang@intel.com>
---
 .../nuplayer/NuPlayerRenderer.cpp                  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
index 4945832..30ed2d8 100644
--- a/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
+++ b/media/libmediaplayerservice/nuplayer/NuPlayerRenderer.cpp
@@ -266,7 +266,7 @@ bool NuPlayer::Renderer::onDrainAudioQueue() {
             CHECK_EQ(mAudioSink->getPosition(&numFramesPlayed), (status_t)OK);
 
             uint32_t numFramesPendingPlayout =
-                mNumFramesWritten - numFramesPlayed;
+                (mNumFramesWritten > numFramesPlayed) ? (mNumFramesWritten - numFramesPlayed) : 0;
 
             int64_t realTimeOffsetUs =
                 (mAudioSink->latency() / 2  /* XXX */
-- 
1.7.4.1

